### ISSUE TITLE
Participate in Web Framework Benchmarks by TechEmpower

### ISSUE THREAD

**Issue by mfirry** [finagle/finch#235]
I'd love to read results of those benchmarks for Finch. [round 10](https://www.techempower.com/benchmarks/#section=data-r10) - [source](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Scala)

**Comment by travisbrown** [comment]
Agreed, this would be great. There was a Finagle entry in a previous round, but it was very broken—I don't know the original contributor, but @stevegury has started repairing it [here](https://github.com/stevegury/FrameworkBenchmarks/commit/facf5ef8a0770d81a56d0b29bb771aeb3ac36d46).

**Comment by stevegury** [comment]
I'm just waiting for the new open-source release of finagle, and I will send a PR.

**Comment by vkostyukov** [comment]
That's a good idea. We can wait for 0.7.0 and send a PR. Although, I believe that it will have the Finagle performance.

**Comment by mfirry** [comment]
Thanks guys!

**Comment by rpless** [comment]
I've started [fixing up](https://github.com/rpless/FrameworkBenchmarks/blob/update-finch/frameworks/Scala/finch/src/main/scala/WebServer.scala) (and soon adding other tests to) the [existing implementation](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Scala/finch) for the benchmarks.

Besides the string encoding issue that we are looking at in #328, any thoughts and comments on the cleanup and additions are welcome.

**Comment by mfirry** [comment]
What's wrong with the existing implementation?

**Comment by rpless** [comment]
Half of the changes are purely stylistic. I've updated from `Endpoint[HttpResponse, HttpRequest]` to `Router[X]`. I've made this change because we are planning on deprecating `Endpoint` per #320. I'm also using implicit conversions of `X` to `HttpResponse`. There's nothing wrong with explicitly returning the Http Response (in a lot of cases you have to), but I think in this case it shows off a some of the power of Finch's Routers.

The other half of the changes are catering to the [Tech Empower tests](http://frameworkbenchmarks.readthedocs.org/en/latest/Project-Information/Framework-Tests/#json-serialization). The tests expect the Http Responses to have their `Server`, `Content-Length`, and `Date` headers set, which Finch does not do by default. I've added a [a filter](https://github.com/rpless/FrameworkBenchmarks/blob/update-finch/frameworks/Scala/finch/src/main/scala/WebServer.scala#L13-L27) that inserts them. The tests also expect Gzip to be turned off, but Finch uses Gzip by default. 

I'm planning on adding the some of the database tests before I actually make a pull request.
Any feedback on the style changes and the filter would be great.

**Comment by mfirry** [comment]
Thanks a lot @rpless . 
I'll keep watching your implementation. 
I'm also pretty eager to see the final results.

I've noticed a lot of Scala implementations there were out-of-date somehow, which was too bad, so I've decided to work on a few (namely finch and http4s).

Side note: I've also worked on [web-frameworks-templates](https://github.com/mfirry/web-frameworks-templates/) which is somehow related but has a different angle (more on the developer feeling rather than performances). @vkostyukov helped me out a bit a few months back.
